src/oidentd.js

```diff
@@ -12,57 +13,31 @@ function OidentdFile(file) {
 }
 
 OidentdFile.prototype = {
-	hookSocket: function(socket, user) {
-		var that = this;
-		var id = null;
+	addSocket: function OidentdFileAddSocket(socket, user) {
+		const id = this.connectionId++;
 
-		socket.on("connect", function() {
-			id = that.addSocket(socket, user);
-			that.refresh();
-		});
-		socket.on("close", function() {
-			that.removeConnection(id);
-			that.refresh();
-		});
-	},
-
-	addSocket: function(socket, user) {
-		var id = this.connectionId++;
 		this.connections[id] = {socket: socket, user: user};
-		return id;
-	},
+		this.refresh();
 
-	removeSocket: function(socket) {
-		for (var id in this.connections) {
-			if (this.connections[id] === socket) {
-				delete this.connections[id];
-				break;
-			}
-		}
+		return id;
 	},
 
-	removeConnection: function(id) {
+	removeSocket: function OidentdFileRemoveSocket(id) {
 		delete this.connections[id];
-	},
 
-	getSockets: function() {
-		return this.connections;
+		this.refresh();
 	},
 
-	refresh: function() {
-		var file = "# Warning: file generated by The Lounge: changes will be overwritten!\n";
+	refresh: function OidentdFileRefresh() {
+		let file = "# Warning: file generated by The Lounge: changes will be overwritten!\n";
 
-		function makeRule(connection) {
-			return "to " + connection.socket.remoteAddress
+		_.forEach(this.connections, (connection) => {
```

Comment:
Reminder as discussed on IRC:

Use [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) for the `connections` object and iterate with native [`forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Iterating_Maps_with_forEach()) (then remove `const _ = require("lodash");` above).